paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java

```diff
@@ -801,6 +806,54 @@ public boolean dropItem(boolean dropAll) {
         // Paper end - Fix HumanEntity#drop not updating the client inv
     }
 
+    @Override
+    public @Nullable Item dropItem(final int slot, final int amount) {
+        // Make sure the slot is in bounds
+        if (slot < 0 || slot >= this.inventory.getSize()) {
+            throw new IndexOutOfBoundsException("Slot " + slot + " out of range for inventory of size " + this.inventory.getSize());
+        }
+
+        return dropItemRaw(this.inventory.getItem(slot), amount, false);
+    }
+
+    @Override
+    public @Nullable Item dropItem(final @NotNull EquipmentSlot slot, final int amount) {
+        return dropItemRaw(this.inventory.getItem(slot), amount, false);
+    }
+
+    @Override
+    public @Nullable Item dropItemRandomly(final int slot, final int amount) {
+        // Make sure the slot is in bounds
+        if (slot < 0 || slot >= this.inventory.getSize()) {
+            throw new IndexOutOfBoundsException("Slot " + slot + " out of range for inventory of size " + this.inventory.getSize());
+        }
+
+        return dropItemRaw(this.inventory.getItem(slot), amount, true);
+    }
+
+    @Override
+    public @Nullable Item dropItemRandomly(final @NotNull EquipmentSlot slot, final int amount) {
+        return dropItemRaw(this.inventory.getItem(slot), amount, true);
+    }
+
+    private @Nullable Item dropItemRaw(final ItemStack originalItemStack, final int amount, final boolean throwRandomly) {
+        if (originalItemStack == null || originalItemStack.isEmpty() || amount <= 0) {
+            return null;
+        }
+
+        final ItemStack clonedItemStack = originalItemStack.clone();
+        final int droppedAmount = Math.min(clonedItemStack.getAmount(), amount);
+        clonedItemStack.setAmount(droppedAmount);
+
+        final ItemEntity droppedEntity = this.getHandle().drop(CraftItemStack.asNMSCopy(clonedItemStack), throwRandomly, true);
+        if (droppedEntity == null) {
+            return null;
+        }
+
+        originalItemStack.setAmount(originalItemStack.getAmount() - droppedAmount);
+        return new CraftItem(this.server, droppedEntity);
```

Comment:
Should just be droppedEntity.getBukkitEntity() instead of constructing a new instance yourself